simple_model_a.py

```diff
@@ -0,0 +1,95 @@
+import matplotlib.pyplot as plt
+import pandas as pd
+from sklearn.metrics import r2_score, mean_squared_error
+"""
+HERE U EXPLAIN WHAT THIS SCRIPT IS FOR. 
+"""
+
+
+def create_simple_ml_model(X_train, X_test, y_train, w_train, reg_):
+    """
+    helper function for make_simple_predictions_for_stacking(X_train, X_test, y_train, w_train, regressors)
+
+    :param X_train: X variables (floats) (training set)
+    :param X_test: X variables (floats) (testing set)
+    :param y_train: endogenous variable (float) (training set)
+    :param w_train: assignment to treatment (0 or 1) (training set)
+    :param reg_: a machine learning model instance
+    :return: the estimated treatment effect for the testing set,
+        as the difference between predicted outcome of treated and predicted outcome of untreated.
+    """
+
+    df = pd.DataFrame(data=X_train)
+    df['assignment'] = w_train
+
+    y_df = pd.DataFrame(data=y_train)
+    reg = reg_.fit(df, y_df)
+
+    X_neg = pd.DataFrame(data=X_test)
+    X_neg['assignment'] = 0
+
+    X_pos = pd.DataFrame(data=X_test)
+    X_pos['assignment'] = 1
+    ret = reg.predict(X_pos) - reg.predict(X_neg)
+    return ret
+
```

Comment:
i'm not sure that there needs to be a 'test set' here, there is no testing going on. i think that the point is to simulate a counterfactual (what would y be if the observation actually got treatment when they really didn't, or vice versa); thats the point of setting the assignment entirely to 1 for one prediction, and entirely to 0 on another prediction.

and in that setup I think it's actually fine to get a model to predict on an observation it has already seen before. it hasn't seen the counterfactual case (for example treated when actually not treated), and it uses what it knows from the treated with similar X's to estimate what the outcome would be if the observation was treated. the difference is the treatment effect. 
so for every observation here, in the case where it is predicting on an observation it has seen before, its fine if it 'overfits' or predicts exactly what it saw, right?

we can leave it in its all the same data with no underlying confounders or anything (at least in most cases from causalml). maybe we can keep this and see how it does with different/difficult propensity scores.